src/gateways/createArchiveLoader.spec.ts

```diff
@@ -246,6 +251,56 @@ describe('archiveLoader', () => {
           expect(error).toBeTruthy();
         }
       });
+
+      it('when the network connection is flaky/offline', async() => {
+        let error: Error | null = null;
+
+        global.fetch = jest.fn(() =>
+          Promise.reject(new TypeError('Failed to fetch'))
+        );
+
+        try {
+          await createArchiveLoader().book('coolid', {
+            booksConfig: { archiveUrl: '/test/archive', books: { coolid: { defaultVersion: 'version' } } },
+          }).load();
+        } catch (e: any) {
+          error = e;
+        }
+
+        if (error) {
+          expect(captureException).not.toHaveBeenCalled();
+          expect((error as ToastMesssageError).messageKey).toEqual(
+            'i18n:notification:toast:archive:load-failure'
+          );
+        } else {
+          expect(error).toBeTruthy();
+        }
+      });
+
+      xit('captures errors if something else went wrong with the fetch', async() => {
```

Comment:
There is an issue with this test not showing that the Sentry call is working. It seems like the Sentry object on the implementation side is not being properly mocked (I've confirmed it is being called with manual testing.) This seems to be a setup/teardown bug, because it is passing in isolation.

Reply:
I've narrowed this down to `resetModules()` in `afterEach`, removing it makes the new specs pass, but breaks the others.

Reply:
Does it work if you take stuff out of the imports at the top of the spec file and add them as requires in the before each?

Reply:
That does work! In hindsight that makes sense, forcing the reload after it gets reset. Thanks!

Reply:
Yea if you don't you get mixed references to globals in the module cache. Usually this happens with react but in this case it might have been sentry